### DISCUSSION TITLE
Request to add an entity that overloads all music players

### DISCUSSION THREAD

**Discussion by ray5378** [music-assistant/support#2352] [Feature requests and ideas]
After shutting down the music player and restarting the mass, the music player will not load at this time. If you open the music player again, it cannot be updated in real-time and needs to be manually reloaded in the webui before it can be used. Please add an entity that reloads all music players to work with homeassistant to automatically make the music player available at any time after booting up,

Written with a translator, if the grammar is incorrect or the expression is unclear, please maintain communication

**Comment by ray5378** [comment]
I have tested the music player and it waited for 5 minutes but couldn't update to the available player in the mass. I don't think it will refresh the available music player, so it can only be manually reloaded in the webui,
Alternatively, setting up real-time refresh of the music player is also possible. There is a homeassistant music player project that can achieve this,
Here is the project address
https://github.com/shaonianzhentan/ha_cloud_music

**Comment by OzGav** [comment]
What music player are you referring to? Restarting Music Assistant should not result in the loss of players

**Reply by ray5378** [reply]
This has nothing to do with what music player to use. When my music player is turned off (completely powered off, not in standby mode) and the mass is restarted, the turned off music player will not be used as an optional player. Of course, this is not a problem

But when I turn on the music player at this time, it won't appear on the available player for a long time, unless you manually reload it on the web page

**Reply by ray5378** [reply]
I am using Plugin providers add Home Assistant MediaPlayers and DLNA try it. but it have same problem,

**Reply by ray5378** [reply]
So I think it's necessary to turn the reload of the music player into a physical entity for automated reloading, or improve the mass program to achieve real-time detection of the availability of the music player

**Reply by OzGav** [reply]
Ok what you are describing is normal. MA doesn’t constantly scan for players. The expectation is you do not keep restarting MA so the players aren’t lost.

**Reply by ray5378** [reply]
My mass is installed in Homeassistant and restarts when updating plugins or systems. However, the smart speaker usually shuts down and loses power, and only turns it on when it needs to be played. I think I'm not the only one using it this way,

So it turns out that as long as you restart the mass or homeassistant, you will have to solve the problem by following these steps because the player cannot be used,

Unlock your phone - Open the homeassistant app - Go to the mass web page and then open the settings page to reload all music players

I think this is a very bad user experience,

**Reply by OzGav** [reply]
MA is in its own docker container, it should be unaffected by restarting HA which is in a different docker container unless you are restarting the host.  What player providers are you using?

**Reply by ray5378** [reply]
Thank you

Yes, just restarting the HA container music player won't disappear. My idea is that in any situation, as long as I open the music player, it can be used without having to solve the problem that it may become unusable in certain situations,

In fact, not only did I completely restart the host, but also during the process of using mass, The mass itself may also experience crashes, and restarting the mass may also cause the player to disappear,
If there is an entity that can be overloaded, it can be easily solved with HA's automation, or the program can be improved so that it can discover available devices in real time

I am using HiVi H5 MKII to add to the haos through the DLNA plugin, and then connect to the music system through the ha plugin of mass

**Reply by OzGav** [reply]
Why aren’t you connecting to this DLNA player directly from MA via the DLNA player provider?

**Reply by ray5378** [reply]
At first, I was accessing DLNA, and the same situation happened. I thought it was a problem with DLNA, so I changed the way I used the HA plugin. I found that there was still a problem, but because there was no difference, I didn't want to change the access method anymore

**Reply by OzGav** [reply]
DLNA as a provider type can be flaky due to the poor implementation of the standard by manufacturers. It is strongly recommended to access the player by the MA native provider.

This request will remain for future consideration.

**Reply by ray5378** [reply]
I will listen to your suggestion and switch to the DLNA music playback that comes with mass

Thank you again for developing such an excellent music software
